setup.py

```diff
@@ -1,4 +1,60 @@
 from setuptools import find_packages, setup
+"""
+Check list to release a new version.
+
+1. Checkout the main branch and pull the latest changes:
+   ```
+   git checkout main
+   git pull
+   ```
+
+2. Checkout a new branch with the version that you'd like to release: v<MINOR-VERSION>-release,
+   for example `v0.5-release`. All patches will be done to that same branch.
+
+3. Update the `__version__` variable in the `src/huggingface_hub/__init__.py` file to point 
+   to the version you're releasing:
+   ```
+   __version__ = "<VERSION>"
+   ```
+   
+4. Make sure that the conda build works correctly by building it locally:
+   ```
+   conda install -c defaults anaconda-client conda-build
+   HUB_VERSION=<VERSION> conda-build .github/conda
+   ```
+
+5. Make sure that the pip wheel works correctly by building it locally and installing it:
+   ```
+   pip install setuptools wheel
+   python setup.py sdist bdist_wheel
+   pip install dist/huggingface_hub-<VERSION>-py3-none-any.whl
+   ```
+   
+6. Commit, tag, and push the branch:
+   ```
+   git commit -am "Release: v<VERSION>"
+   git tag v<VERSION> -m "Adds tag v<VERSION> for pypi and conda"
+   git push -u --tags origin v<MINOR-VERSION>-release
+   ```
```

Comment:
to give visibility to others, this can be a PR to the release branch instead of direct commit.

Reply:
It would also help with checking the doc build on that PR.

Reply:
How would you name the branch from which that PR is made? Others will get visibility with the release already, I am not 100% sure I see why the PR is necessary here

Reply:
that PR comes from the maintainer's own fork, doesn't matter how it's named. E.g: https://github.com/scikit-learn/scikit-learn/pull/21120

In most cases it'll be a simple PR, but sometimes you might want to include more commits from the `main` branch.